webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -182,23 +205,171 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
   const logFilters = (
     <LogViewerSelect
       options={filterOptions}
-      showSearch={true}
+      searchOn={searchOn}
+      showSearch={false}
       values={filterValues}
       onChange={handleFilterChange}
+      onClickSearch={() => setSearchOn((prev) => !prev)}
       onReset={handleFilterReset}
     />
   );
 
+  const throttledChangeSearch = useMemo(
+    () =>
+      throttle(
+        500,
+        (s: string) => {
+          updateSettings({ ...settings, searchText: s });
+        },
+        { noLeading: true },
+      ),
+    [updateSettings, settings],
+  );
+
+  useEffect(() => {
+    return () => {
+      throttledChangeSearch.cancel();
+    };
+  }, [throttledChangeSearch]);
+
+  const onSearchChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchInput(e.target.value);
+      throttledChangeSearch(e.target.value);
+      if (!e.target.value) {
+        setSearchResults([]);
+        setSelectedLog(undefined);
+      }
+    },
+    [throttledChangeSearch],
+  );
+
+  const formatedSearchResults = useMemo(() => {
+    const key = settings.searchText;
+
+    if (!key) return;
+    const formated: TrialLog[] = [];
+    searchResults.forEach((l) => {
+      const content = l.log;
+      if (!content) return;
+      if (settings.enableRegex) {
+        try {
+          new RegExp(key);
+        } catch {
+          return;
+        }
+      }
+
+      const i = settings.enableRegex ? content.match(`${key}`)?.index : content.indexOf(key);
+      if (!i || i < 0) return;
+      const keyLen = settings.enableRegex ? content.match(`${key}`)?.[0].length || 0 : key.length;
+      const j = i + keyLen;
+      const numOfChar = Math.floor(searchWidth / 18) - keyLen;
+      let start = Math.max(i - numOfChar, 0);
+      let end = Math.min(j + numOfChar, content.length);
+      if (start > 0 && end === content.length) {
+        start = Math.max(start - (numOfChar - end + j), 0);
+      } else if (start === 0 && end < content.length) {
+        end = Math.min(end + numOfChar - i + start, content.length);
+      }
+      formated.push({
+        ...l,
+        log: `${start > 0 ? '...' : ''}${content.slice(start, i)}<span style="background-color: #E7F7FF">${content.slice(i, j)}</span>${content.slice(j, end)}${end < content.length ? '...' : ''}`,
```

Comment:
can we move the color highlight to css so it's consistent with the rest of our codebase? It'd also allow us to have different colors for light mode vs dark mode.